app/services/SalesforceService.scala

```diff
@@ -5,26 +5,39 @@ import com.gu.memsub.NormalisedTelephoneNumber
 import com.gu.salesforce.ContactDeserializer.Keys
 import com.gu.salesforce._
 import com.typesafe.scalalogging.LazyLogging
-import model.{PaperData, PersonalData}
+import model.{DeliveryRecipient, PaperData, PersonalData, PurchaserIdentifiers}
 import play.api.libs.json.{JsObject, Json}
 
 import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 
 trait SalesforceService extends LazyLogging {
   def repo: SimpleContactRepository
-  def createOrUpdateUser(personalData: PersonalData, paperData: Option[PaperData], userId: Option[IdMinimalUser]): Future[ContactId]
+  def createOrUpdateBuyerAndRecipient(personalData: PersonalData, paperData: Option[PaperData], userId: Option[IdMinimalUser]): Future[PurchaserIdentifiers]
   def isAuthenticated: Boolean
 }
 
 class SalesforceServiceImp(val repo: SimpleContactRepository) extends SalesforceService {
-  override def createOrUpdateUser(personalData: PersonalData, paperData: Option[PaperData], userId: Option[IdMinimalUser]): Future[ContactId] =
-    repo.upsert(userId.map(_.id), SalesforceService.createSalesforceUserData(personalData, paperData))
+  override def createOrUpdateBuyerAndRecipient(personalData: PersonalData, paperData: Option[PaperData], userId: Option[IdMinimalUser]): Future[PurchaserIdentifiers] = {
+    for {
+      buyerContact <- repo.upsert(userId.map(_.id), SalesforceService.getJSONForBuyerContact(personalData, paperData))
+      recipientContact <- { // Only create a Delivery / Recipient (Related) Contact if the recipient is a Giftee
```

Comment:
We could make this clearer (and testable) by pulling the code which decides to create a recipient into a separate function?

Reply:
I would add some logging here too

Reply:
Agreed, this is better off being in a separate function. I'll raise a separate PR later today.